src/rust/src/wallet.rs

```diff
@@ -292,27 +412,34 @@ pub extern "C" fn orchard_wallet_free(wallet: *mut Wallet) {
 }
 
 #[no_mangle]
-pub extern "C" fn orchard_wallet_checkpoint(wallet: *mut Wallet) {
+pub extern "C" fn orchard_wallet_checkpoint(wallet: *mut Wallet, block_height: u32) -> bool {
     let wallet = unsafe { wallet.as_mut() }.expect("Wallet pointer may not be null");
-    wallet.checkpoint_witness_tree();
+    wallet.checkpoint(block_height.into())
 }
 
 #[no_mangle]
-pub extern "C" fn orchard_wallet_rewind(wallet: *mut Wallet) -> bool {
+pub extern "C" fn orchard_wallet_is_checkpointed(wallet: *const Wallet) -> bool {
+    let wallet = unsafe { wallet.as_ref() }.expect("Wallet pointer may not be null");
+    wallet.is_checkpointed()
+}
+
+#[no_mangle]
+pub extern "C" fn orchard_wallet_rewind(wallet: *mut Wallet, to_height: BlockHeight) -> bool {
     let wallet = unsafe { wallet.as_mut() }.expect("Wallet pointer may not be null");
-    wallet.rewind_witness_tree()
+    wallet.rewind(to_height)
 }
 
 #[no_mangle]
 pub extern "C" fn orchard_wallet_add_notes_from_bundle(
     wallet: *mut Wallet,
+    block_height: u32,
```

Comment:
This similarly needs to be optional, probably by making it a `*const u32` and treating `nullptr` as `None`.